### PR TITLE
ref(members): Convert member settings to FC

### PR DESCRIPTION
**Pull request by scttcper**
Removes async component, uses a few mutations

### CODE REVIEW

File: static/app/views/settings/organizationMembers/organizationMemberDetail.tsx [hunk 1]
```diff
@@ -255,175 +244,178 @@ class OrganizationMemberDetail extends DeprecatedAsyncView<Props, State> {
...
+  function renderMemberStatus() {
+    if (memberDeactivated) {
       return (
         <em>
           <DisabledMemberTooltip>{t('Deactivated')}</DisabledMemberTooltip>
         </em>
       );
     }
-    if (member.expired) {
+    if (member!.expired) {
```
Comment (priscilawebdev) [inline static/app/views/settings/organizationMembers/organizationMemberDetail.tsx:267]: do we need `!` here? in the line 255 we return `<NotFound />` if it is not available

File: static/app/views/settings/organizationMembers/organizationMemberDetail.tsx [hunk 2]
```diff
@@ -45,165 +55,149 @@ const TWO_FACTOR_REQUIRED = t(
...
       });
-      this.setState({
-        member: updatedMember,
-        orgRole: updatedMember.orgRole,
-        teamRoles: updatedMember.teamRoles,
-        busy: false,
-      });
+    },
+    onMutate: () => {
+      addLoadingMessage(t('Saving...'));
```
Comment (priscilawebdev) [inline static/app/views/settings/organizationMembers/organizationMemberDetail.tsx:106]:
```suggestion
      addLoadingMessage(t('Saving\u2026'));
```

File: static/app/views/settings/organizationMembers/organizationMemberDetail.tsx [hunk 3]
```diff
@@ -45,165 +55,149 @@ const TWO_FACTOR_REQUIRED = t(
...
-      ) ?? [];
-
-    try {
-      await Promise.all(requests);
-      router.push(normalizeUrl(`/settings/${organization.slug}/members/`));
+  );
+
+  const {mutate: reset2fa, isPending: isResetting2fa} = useMutation<unknown>({
+    mutationFn: () => {
+      const {user} = member!;
```
Comment (priscilawebdev) [inline static/app/views/settings/organizationMembers/organizationMemberDetail.tsx:149]: can 'member' be not defined here? shall we check for it?

File: static/app/views/settings/organizationMembers/organizationMemberDetail.tsx [hunk 4]
```diff
@@ -255,175 +244,178 @@ class OrganizationMemberDetail extends DeprecatedAsyncView<Props, State> {
...
+        helpText={
+          isPartnershipUser
+            ? t('You cannot make changes to this partner-provisioned user.')
+            : undefined
+        }
+      />
+
+      <Teams slugs={member.teams}>
+        {({initiallyLoaded}) => (
+          <Fragment>
```
Comment (priscilawebdev) [inline static/app/views/settings/organizationMembers/organizationMemberDetail.tsx:388]: do we need `Fragment` here?

File: static/app/views/settings/organizationMembers/organizationMemberDetail.tsx [hunk 5]
```diff
@@ -45,165 +55,149 @@ const TWO_FACTOR_REQUIRED = t(
...
-    teamRoles[i] = {...teamRoles[i], role};
-    this.setState({teamRoles});
+    newTeamRoles[i] = {...newTeamRoles[i], role};
+    setTeamRoles(newTeamRoles);
   };
 
-  showResetButton = () => {
-    const {organization} = this.props;
-    const {member} = this.state;
+  const showResetButton = () => {
```
Comment (priscilawebdev) [inline static/app/views/settings/organizationMembers/organizationMemberDetail.tsx:200]: do we want to use `useCallback` for these functions?

Comment (scttcper) [reply static/app/views/settings/organizationMembers/organizationMemberDetail.tsx]: reduced a few functions, but i'm not super worried about rerendering here

**Review by malwilley — APPROVED** [review]

**Pull request merged**